core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -230,24 +230,28 @@ public void applyTo(TableMetadata.Builder metadataBuilder) {
 
   class SetSnapshotRef implements MetadataUpdate {
     private final String name;
-    private final long snapshotId;
+    private final SnapshotRef ref;
```

Comment:
The [`SetSnapshotRef` update](https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L1233-L1242) in the REST spec has all of the fields here, rather than holding a `SnapshotRef` object.

To make this serialize like the spec, this should have `type`, `maxRefAgeMs`, `maxSnapshotAgeMs`, and `minSnapshotsToKeep` fields.

Reply:
Thanks for pointing to the spec, will keep that in mind and update.